inference/models/transformers/transformers.py

```diff
@@ -0,0 +1,213 @@
+import os
+import re
+
+import numpy as np
+from peft import LoraConfig, get_peft_model
+from PIL import Image
+from transformers import AutoModel, AutoProcessor, PaliGemmaForConditionalGeneration
+
+from inference.core.env import HUGGINGFACE_TOKEN, MODEL_CACHE_DIR
+
+cache_dir = os.path.join(MODEL_CACHE_DIR)
+import os
+import time
+from time import perf_counter
+from typing import Any, List, Tuple, Union
+
+import torch
+from PIL import Image
+
+from inference.core.cache.model_artifacts import save_bytes_in_cache
+from inference.core.entities.requests.inference import LMMInferenceRequest
+from inference.core.entities.responses.inference import (
+    InferenceResponseImage,
+    LMMInferenceResponse,
+)
+from inference.core.env import API_KEY, MODEL_CACHE_DIR, PALIGEMMA_VERSION_ID
+from inference.core.exceptions import ModelArtefactError
+from inference.core.logger import logger
+from inference.core.models.base import PreprocessReturnMetadata
+from inference.core.models.roboflow import RoboflowInferenceModel
+from inference.core.roboflow_api import (
+    ModelEndpointType,
+    get_from_url,
+    get_roboflow_model_data,
+)
+from inference.core.utils.image_utils import load_image_rgb
+
+DEVICE = "cuda:0" if torch.cuda.is_available() else "cpu"
```

Comment:
What if we would like to run on 2nd GPU?

Reply:
We can load from environment variable I guess?